privacyidea/lib/utils/__init__.py

```diff
@@ -1137,13 +1145,10 @@ def check_pin_policy(pin, policy):
 
     if policy[0] == "+":
         # grouping
-        necessary = []
         for char in policy[1:]:
-            necessary.append(chars.get(char))
-        necessary = "|".join(necessary)
```

Comment:
the OR connection was not correct here, since we want to have all the string-lists included in the password.

Reply:
According to the documentation, this change is wrong. I'll revert it.

Reply:
done